src/pages/requests/index.tsx

```diff
@@ -74,84 +84,96 @@ export function Requests() {
+  const tableDataByCategories = groupedByCategories.map(groupedByCategoriesToTableData);
 
+  // TODO: move this logic to map component - it should get filters via context and process them accordingly
+  // TODO: fix filter mapping - it should use category_id
+  // TODO: add some comments / typing explaining what kind of black the magic is happening here :)
   const layerFilterCategory = activeCategoryFilters.length
     ? ["in", ["get", "category"], ["array", ["literal", activeCategoryFilters]]]
     : ["==", ["get", "category"], "ALL"];
 
   const layerFilterDate = activeDateFilter ? ["==", ["get", "date"], ["string", activeDateFilter]] : ["boolean", true];
-
   const layerFilter = ["all", layerFilterCategory, layerFilterDate];
 
-  const selectedDate = activeDateFilter;
-  const activeCategories = activeCategoryFilters;
-  const categoriesTableData = mapLocationsToTableData(decodedAndGroupedByCategory);
-  const filteredAndGroupedByCategory = categoriesTableData
-    .map((data: CategoriesTableData) => {
-      return {
-        ...data,
-        hidden: data.hidden.filter((req) => req.date === selectedDate),
-      };
-    })
-    .filter((data: CategoriesTableData) => (activeCategories.length ? activeCategories.some((category) => data.name === category) : data)) // TODO: activeCategories needs to be full for "ALL"
-    .map((data: CategoriesTableData) => {
-      const getTotalForCategory = () => data.hidden.reduce((partialSum, aidRequest) => partialSum + aidRequest.total, 0);
-      return {
-        ...data,
-        total: getTotalForCategory(),
-      };
-    })
-    .filter((data: CategoriesTableData) => data.total !== 0)
-    .sort((a: any, b: any) => b.total - a.total);
-
-  const locationsTableData = mapCategoriesToTableData(decodedAndGroupedByLocation);
-  const filteredAndGroupedByLocation = locationsTableData
-    .map((data: LocationsTableData) => {
-      return {
-        ...data,
-        decodedAidRequests: data.decodedAidRequests.filter((req) => req.date === selectedDate),
-      };
-    })
-    .map((data: LocationsTableData) => {
-      return {
-        ...data,
-        hidden: data.decodedAidRequests
-          .map((req) => ({ name: req.name, total: req.amount }))
-          .filter((req) => (activeCategories.length ? activeCategories.some((category) => req.name === category) : req)), // TODO: activeCategories needs to be full for "ALL"
-      };
-    })
-    .map((data: LocationsTableData) => {
-      const getTotalForLocation = () => data.hidden.reduce((partialSum, aidRequest) => partialSum + aidRequest.total, 0);
-      return {
-        ...data,
-        total: getTotalForLocation(),
-      };
-    })
-    .filter((data: LocationsTableData) => data.total !== 0)
-    .sort((a: any, b: any) => b.total - a.total);
+  const { selectedTabId, setSelectedTabId } = useSidebarContext();
 
-  if (!cities) {
-    return <Layout header={<Header />}>{/* <Loader /> */}</Layout>;
-  }
+  // TODO: implement proper loaders
```

Comment:
++ 

Made a related comment on https://github.com/ukraine-taskforce/requests-dashboard/pull/40#discussion_r828053945

Reply:
Yeah, and I think `useQuery` supports loading state out of the box.